pkg/controller/storageprofile-controller.go

```diff
@@ -60,22 +61,22 @@ func (r *StorageProfileReconciler) reconcileStorageProfile(sc *storagev1.Storage
 	storageProfile.Status.Provisioner = &sc.Provisioner
 	storageProfile.Status.CloneStrategy = storageProfile.Spec.CloneStrategy
 
-	storageProfile.Status.ClaimPropertySets = storageProfile.Spec.ClaimPropertySets
+	var claimPropertySets []cdiv1.ClaimPropertySet
 
-	var claimPropertySet *cdiv1.ClaimPropertySet
-	// right now the CDI supports only a single property set
-	if len(storageProfile.Status.ClaimPropertySets) > 0 {
-		claimPropertySet = &storageProfile.Status.ClaimPropertySets[0]
+	if len(storageProfile.Spec.ClaimPropertySets) > 0 {
+		for _, cps := range storageProfile.Spec.ClaimPropertySets {
+			if len(cps.AccessModes) == 0 && cps.VolumeMode != nil {
+				err = fmt.Errorf("must provide access mode for volume mode: %s", *cps.VolumeMode)
+				log.Error(err, "Unable to update StorageProfile")
+				return reconcile.Result{}, err
+			}
+		}
+		claimPropertySets = storageProfile.Spec.ClaimPropertySets
```

Comment:
so this is either all set by user or from CDI knowledge, in my opinion this is good and simple to understand